action/staking/stake.go

```diff
@@ -114,6 +115,10 @@ func (s stakeTx) Validate(ctx *action.Context, tx action.SignedTx) (bool, error)
 	if err != nil {
 		return false, action.ErrNotEnoughFund
 	}
+
```

Comment:
Need to move this state-dependant check to func 'runCheckStake'.

Reply:
You can either move this check to runCheckStake or just remove it, since 'MinusFromAddress' will return you an error in case of insufficient balance.